### PR TITLE
Account Settings: Fix Overlap with Profile Links on Mobile

### LINKED ISSUE #26257: Account Settings: Help icon obscures buttons in personal profile section
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/me on mobile
2. In Profile Links, tap Add and tap Add WordPress site
3. Select a site/sites in your account you wish to add
4. Scroll down to the Add button

#### What I expected
To see the Add Button without any other elements overlapping

#### What happened instead
The Help icon is overlapping the Add button. It doesn't make the button unusable, but is not a great experience. Likewise, after pressing Add, the x icon to remove the site at the bottom of the list is partially obscured.

#### Browser / OS version
Confirmed in Chrome on Android

#### Screenshot / Video
screenshot_20180723-152858
screenshot_20180723-153059

#### Context / Source

Reported via [Twitter](https://twitter.com/FairWitness1/status/1021375529658798080)

### PR DESCRIPTION
**Pull request by torres126**
#### Changes proposed in this Pull Request

Adds some padding to the bottom of the "Profile Links" box on mobile devices, so that the inline help icon no longer overlaps with buttons. 

#### Testing instructions

Ensure that there's no overlaps and no unintended changes at https://wordpress.com/me, meaning you can click all buttons. 

**Current:**

beforedgfdfgfgdfgd

beforefsdgfdfdggdf

**Intended Proposal:**

after_sgfdgfdgfs

after_dfgdfggdf

**After when an error appears:**

with_an_error_dfgfddgfgdf

(cc @ebinnion, @dmsnell & @tyxla) 

Fixes #26257

### CODE REVIEW

File: client/me/profile-links/index.jsx [hunk 1]
```diff
@@ -168,7 +168,7 @@ class ProfileLinks extends React.Component {
 
 	render() {
 		return (
-			<div>
+			<div className="me-profile-links">
```
Comment (tyxla) [inline client/me/profile-links/index.jsx]: This classname doesn't seem right. As per our [classname recommendations](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#classes), it should have the component name as a prefix (which in our case is `profile-links__`), and then have some sort of suffix - if that's the main wrapper, we often add `main` as a suffix.

File: client/me/profile-links/style.scss [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/** @format */
```
Comment (tyxla) [inline client/me/profile-links/style.scss]: I think the Prettier pragmas are no longer necessary after #29403 (also see #29404)

**Comment by torres126** [comment]
Thanks! Updated the class name and removed the Prettier pragma. I tested a few places across the Profile, Reader and My Sites section, but this seems to be the only one where at the furthest point of possible scrolling, the inline help button is overlapping something significant, such as the buttons here. I might be missing somewhere though.

**Comment by tyxla** [comment]
Oh well, I'm certain there will be more cases like this. A quick test I did showed some:

**Sharing settings page on Galaxy S5:**


**Comments page on iPhone 5/SE:**


Seems like we need a universal solution, which is why I recommended asking for design feedback.

**Comment by torres126** [comment]
Ahh, got it. Just to confirm, who should be pinged for design feedback? (I thought it was a label :sweat_smile: )

**Comment by tyxla** [comment]
I'm sure anyone from @Automattic/design would be happy to chime in with feedback on this one 😉

**Comment by folletto** [comment]
My suggestion for a _quick fix_ would be to add padding at the end of the page instead at the end of cards. Changing the padding inside a card would modify the proportions of the card thus breaking both the balance of the card itself, as well as the standard padding for the cards.

**Comment by tyxla** [comment]
Or perhaps instead of blank space at the bottom we could use the list end component that we display the bottom of the posts page:



or something similar.

**Comment by torres126** [comment]
Hi @tyxla, I found this PR again on my list of PRs and I think I've caught up with this again (and thankfully, my knowledge of things has increased a lot more since I first opened this PR!). 

Would you mind giving this another spin now?

Screenshot 2019-04-03 at 18 08 44

I quite favour the List End approach, it removes the requirement for additional styling and fixes the issue.

File: client/me/profile-links/index.jsx [hunk 3]
```diff
@@ -173,20 +172,23 @@ class ProfileLinks extends React.Component {
...
+						<AddProfileLinksButtons
+							showingForm={ !! this.state.showingForm }
+							onShowAddOther={ this.showAddOther }
+							showPopoverMenu={ this.state.showPopoverMenu }
+							onShowAddWordPress={ this.showAddWordPress }
+							onShowPopoverMenu={ this.showPopoverMenu }
+							onClosePopoverMenu={ this.closePopoverMenu }
+						/>
+					</SectionHeader>
+					<Card>{ !! this.state.showingForm ? this.renderForm() : this.renderProfileLinks() }</Card>
```
Comment (tyxla) [inline client/me/profile-links/index.jsx]: This `!!` seems to be unnecessary.

File: client/me/profile-links/index.jsx [hunk 4]
```diff
@@ -173,20 +172,23 @@ class ProfileLinks extends React.Component {
...
-						showPopoverMenu={ this.state.showPopoverMenu }
-						onShowAddWordPress={ this.showAddWordPress }
-						onShowPopoverMenu={ this.showPopoverMenu }
-						onClosePopoverMenu={ this.closePopoverMenu }
-					/>
-				</SectionHeader>
-				<Card>{ this.state.showingForm ? this.renderForm() : this.renderProfileLinks() }</Card>
-			</div>
+			<Fragment>
+				<div>
```
Comment (tyxla) [inline client/me/profile-links/index.jsx]: This `<div />` seems to be unnecessary and can be removed.

File: client/me/profile-links/index.jsx [hunk 5]
```diff
@@ -173,20 +172,23 @@ class ProfileLinks extends React.Component {
...
+					<QueryProfileLinks />
+					<SectionHeader label={ this.props.translate( 'Profile Links' ) }>
+						<AddProfileLinksButtons
+							showingForm={ !! this.state.showingForm }
```
Comment (tyxla) [inline client/me/profile-links/index.jsx]: Is there any benefit from this `!!`?

Comment (torres126) [reply client/me/profile-links/index.jsx]: Not sure how that got there tbh. 🤔 

Oh well, nice spot, addressed in the last commit!

File: client/me/profile-links/index.jsx [hunk 6]
```diff
@@ -173,20 +172,21 @@ class ProfileLinks extends React.Component {
...
-						showPopoverMenu={ this.state.showPopoverMenu }
-						onShowAddWordPress={ this.showAddWordPress }
-						onShowPopoverMenu={ this.showPopoverMenu }
-						onClosePopoverMenu={ this.closePopoverMenu }
-					/>
-				</SectionHeader>
-				<Card>{ this.state.showingForm ? this.renderForm() : this.renderProfileLinks() }</Card>
-			</div>
+					<SectionHeader label={ this.props.translate( 'Profile Links' ) }>
+						<AddProfileLinksButtons
```
Comment (dmsnell) [inline client/me/profile-links/index.jsx]: Looks like some extraneous indentation snuck in here. Have you been able to get the git hooks setup @torres126? They should auto-format on save.

Comment (torres126) [reply client/me/profile-links/index.jsx]: Hmm, nope, I'm not aware what that is. I'll look into that when I get the chance. :) 

(Before a few days ago, my PRs were made on my tablet which kept erasing lines and I couldn't use tab, so I think many of my PRs may have some odd spacing as a result too)

Comment (dmsnell) [reply client/me/profile-links/index.jsx]: I'm not sure why they aren't working as they should be Automattic. The hooks are in `.git/hooks`

Can you verify that they are executable?
Are you on Windows?
Are you committing via the `git` command line utility or using some GUI?

Comment (torres126) [reply client/me/profile-links/index.jsx]: > Can you verify that they are executable?

Me isn't sure how to do that :sweat_smile: 

> Are you on Windows?

Android for all my previous PRs until a few days ago, where I'm on Mac

> Are you committing via the git command line utility or using some GUI?

Directly on github.com

Comment (dmsnell) [reply client/me/profile-links/index.jsx]: That makes sense then if you've been editing code through the browser. As you get into a workflow on macOS the auto-formatting should run on `git commit`

**Pull request merged**